torch/_dynamo/eval_frame.py

```diff
@@ -643,6 +645,177 @@ def __bool__(self):
+            )
+            assert_msg = f"Input #{self.count}'s dimension #{constraint.dim} size is outside of supported dynamic range"
+
+            if min_int_val and min_int_val > 2:
+                gt = self.tracer.create_proxy(
+                    "call_function", operator.gt, (dim, min_int_val), {}
+                )
+                tensor_gt = self.tracer.create_proxy(
+                    "call_function", torch.scalar_tensor, (gt,), {}
+                )
+                self.tracer.create_proxy(
+                    "call_function",
+                    torch._assert_async,
+                    (tensor_gt, assert_msg),
+                    {},
+                )
+
+            if max_int_val and max_int_val < math.inf:
+                lt = self.tracer.create_proxy(
+                    "call_function", operator.lt, (dim, max_int_val), {}
+                )
+                tensor_lt = self.tracer.create_proxy(
+                    "call_function", torch.scalar_tensor, (lt,), {}
+                )
+                self.tracer.create_proxy(
+                    "call_function",
+                    torch._assert_async,
+                    (tensor_lt, assert_msg),
+                    {},
+                )
+
+        self.count += 1
+        return arg
+
+    def call_function(self, target, args, kwargs):
+        arg = super().call_function(target, args, kwargs)
+        from torch._export.constraints import constrain_as_size
+
+        if target == constrain_as_size:
+            min_val = kwargs["min"]
+            max_val = kwargs["max"]
+            gt = self.tracer.create_proxy(
+                "call_function", operator.gt, (args[0], min_val), {}
+            )
+            tensor_gt = self.tracer.create_proxy(
+                "call_function", torch.scalar_tensor, (gt,), {}
+            )
+            lt = self.tracer.create_proxy(
+                "call_function", operator.lt, (args[0], max_val), {}
+            )
+            tensor_lt = self.tracer.create_proxy(
+                "call_function", torch.scalar_tensor, (lt,), {}
+            )
+
+            self.tracer.create_proxy(
+                "call_function",
+                torch._assert_async,
+                (tensor_gt, "min_val"),
+                {},
+            )
+
+            self.tracer.create_proxy(
+                "call_function",
+                torch._assert_async,
+                (tensor_lt, "max_val"),
+                {},
+            )
+        return arg
+
+    def run_node(self, n):
+        self.current_node = n
+        r = super().run_node(n)
+        if "val" in self.current_node.meta:
+            r.node.meta["val"] = self.current_node.meta["val"]
+        if "tensor_dict" in self.current_node.meta:
+            r.node.meta["tensor_dict"] = self.current_node.meta["tensor_dict"]
+        return r
+
+
+class _ChangeInputOutputSignature(torch.fx.interpreter.Transformer):
```

Comment:
Some comments would be nice

Reply:
This was here before my change. I am not actually sure what it does...